npeps/npep-173-service-account-matches.md

```diff
@@ -0,0 +1,51 @@
+# NPEP-173: NPEP template
+
+* Issue: [#173](https://github.com/kubernetes-sigs/network-policy-api/issues/173)
+* Status: Provisional
+
+## TLDR
+
+Ability to express subjects and peers in (Baseline)AdminNetworkPolicy API
+using [Kubernetes service accounts](https://kubernetes.io/docs/concepts/security/service-accounts/).
+
+## Goals
+
+* Implement service account as a way to select pods in the policy's subject
+* Implement service account as a way to select pods in the policy's peers
+
+## Non-Goals
+
+* There might be other identity management constructs like SPIFFE IDs which are outside the scope of this enhancement. We can only provide a way to select constructs known to core Kubernetes.
```

Comment:
Implementor bottlenecks - since implementations might not want to consume the whole CRD...

Reply:
there was a conclusion at kubecon that it makes more sense for it to be a new CRD?

Reply:
what about L7 layer CRD instead of the L3/L4 matches since if its in the policy its usually for L4/L3 IP based matches

Reply:
The typical service mesh pattern is required/expected identity-based authorization policy at L4 (default deny, allow TCP connection based on identity only if a lower level like CNI did not already drop/block the attempt at L3), with optional support for additional constraints at L7 (such as only allowing requests from a given identity on specific paths).

Reply:
thanks @mikemorris for the insights here! very valuable, I have tried to capture some parts of this point and the point @danwinship mentioned above under "unresolved discussions" so that its written down somewhere..